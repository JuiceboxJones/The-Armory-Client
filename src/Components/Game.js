import React from "react";
import { Link } from "react-router-dom";

export default function Game(props) {
  let data = props.props;
  console.log(data.title)
  return (
    <div className='game-item-container'>
    <Link to={`/games/${data.id}`}>
    <div className='game-img'>
      <img src={data.image_url} alt="Game Cover" />
    </div>
      <h3>{data.title}</h3>
    </Link>
    {data.tags.map((tag, i) => {
      return <span className='small-detail' key={i}>{tag}</span>;
    })}
    <p>Available Parties: {data.party_count}</p>
  </div>
  )
}

