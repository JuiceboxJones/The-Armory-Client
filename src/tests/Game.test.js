import React from "react";
import Game from "../Components/Game";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
// import { render } from 'react-testing-library';

const data = {
  id: "aa0e8ce9-1a71-42e7-804d-6838556fa6ed",
  title: "Overwatch",
  image_url:
    "https://res.cloudinary.com/squadarmory/image/upload/v1557172650/overwatch_epjgky.png",
  tags: ["Shooter", "FPS"]
};

describe("Game component", () => {
  // TypeError: Cannot read property 'id' of undefined
  it("renders without crashing", () => {
    console.log(data.id);
    shallow(<Game props={data} />);

  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Game props={data}/>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
