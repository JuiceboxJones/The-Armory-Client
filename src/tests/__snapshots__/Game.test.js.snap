// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component renders the UI as expected 1`] = `
<div
  className="game-item-container"
>
  <a
    href="/games/aa0e8ce9-1a71-42e7-804d-6838556fa6ed"
    onClick={[Function]}
  >
    <div
      className="game-img"
    >
      <img
        alt="Game Cover"
        src="https://res.cloudinary.com/squadarmory/image/upload/v1557172650/overwatch_epjgky.png"
      />
    </div>
    <h3>
      Overwatch
    </h3>
  </a>
  <span
    className="small-detail"
  >
    Shooter
  </span>
  <span
    className="small-detail"
  >
    FPS
  </span>
  <p>
    Available Parties: 
  </p>
</div>
`;
